src/mlnet/Telemetry/Telemetry.cs

```diff
@@ -1,144 +1,239 @@
+using System.Linq;
+using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.ApplicationInsights;
+using Microsoft.DotNet.Cli.Telemetry;
 using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Configurer;
 using Microsoft.DotNet.PlatformAbstractions;
+using Microsoft.ML.AutoML;
 
-namespace Microsoft.DotNet.Cli.Telemetry
+namespace Microsoft.ML.CLI.Telemetry
 {
-    public class Telemetry : ITelemetry
+    /// <summary>
+    /// Houses CLI telemetry collection and utility methods.
+    /// </summary>
+    internal static class Telemetry
     {
-        private TelemetryClient _client = null;
-        private Dictionary<string, string> _commonProperties = new Dictionary<string, string>();
-        private Task _trackEventTask = null;
+        private static TelemetryClient _client;
+        private static Dictionary<string, string> _commonProperties;
+        private static bool _enabled;
+        private static Task _initializationTask;
+        private static int _outstandingTelemetryEventCount;
 
         private const string InstrumentationKey = "c059917c-818d-489a-bfcb-351eaab73f2a";
         private const string MlTelemetryOptout = "MLDOTNET_CLI_TELEMETRY_OPTOUT";
         private const string MachineId = "MachineId";
-
-        public bool Enabled { get; }
         
-        public Telemetry()
+        public static void Initialize()
         {
             var optedOut = Env.GetEnvironmentVariableAsBool(MlTelemetryOptout, false);
-
-            Enabled = !optedOut;
-
-            if (!Enabled)
+            _enabled = !optedOut;
+            if (!_enabled)
             {
                 return;
             }
-          
-            //initialize in task to offload to parallel thread
-            _trackEventTask = Task.Factory.StartNew(() => InitializeTelemetry());
+
+            // Initialize in task to offload to parallel thread
+            _initializationTask = Task.Factory.StartNew(() => InitializeTask());
         }
 
-        public void TrackEvent(
-            string eventName, 
+        /// <summary>
+        /// Send telemetry event.
+        /// </summary>
+        public static void TrackEvent(
+            string eventName,
             IDictionary<string, string> properties,
-            IDictionary<string, double> measurements)
+            TimeSpan? duration = null,
+            Exception ex = null)
         {
-            if (!Enabled)
+            if (!_enabled)
             {
                 return;
             }
-
-            //continue task in existing parallel thread
-            _trackEventTask = _trackEventTask.ContinueWith(
-                x => TrackEventTask(eventName, properties, measurements)
-            );
+            // Increment count of outstanding telemetry events.
+            Interlocked.Increment(ref _outstandingTelemetryEventCount);
+            _initializationTask.ContinueWith(x => TrackEventTask(eventName, properties, duration, ex));
```

Comment:
since the task is used for not just initialization, the _intializationTask name is a bit decieving

Reply:
Is anything else aside from initialization being done in the initialization task?